### PR TITLE
Make `create_image_fetcher` and `image_fetcher` private.

### PR DESCRIPTION
**Pull request by mkcor**
## Description

Supersedes #6855 -- although, @seanquinn appears to be a first-time contributor with #6855, so I would have really preferred it if it could have been reviewed/approved/merged before #6886! :disappointed: 

I thought I would update the release notes separately, as in #6949?

I updated the conflicting file following the logic implemented in #6886, but one thing bugged me (@lagru, please see inline comment).

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- Descriptive commit messages (see below)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
  `doc/release/release_dev.rst`.
- There is a bot to help automate backporting a PR to an older branch. For
  example, to backport to v0.19.x after merging, add the following in a PR
  comment: `@meeseeksdev backport to v0.19.x`
- To run benchmarks on a PR, add the `run-benchmark` label. To rerun, the label
  can be removed and then added again. The benchmark output can be checked in
  the "Actions" tab.

### CODE REVIEW

File: skimage/data/_fetchers.py [hunk 1]
```diff
@@ -212,7 +212,7 @@ def _fetch(data_filename):
         return legacy_file_path
 
     # Case 3: file is not present locally
-    if image_fetcher is None:
+    if _image_fetcher is None:
```
Comment (mkcor) [inline skimage/data/_fetchers.py:215]: @lagru we just checked for the same condition (line 198), so shouldn't this block go under the first `if _image_fetcher is None:` conditional?

Comment (lagru) [reply skimage/data/_fetchers.py:215]: We want to hit case 1 and 2 first, so the content of this block should definitely come after. And `_create_image_fetcher` populates `data_dir` currently in a weird way, requiring us to select the parent directory only if `_image_fetcher is None`. I thought about refactoring this earlier, but I think I wanted to keep the scope of that previous PR small as it was already taking long enough to get the fix in. You could try updating that if you want...

Comment (mkcor) [reply skimage/data/_fetchers.py:215]: I see, thanks for explaining. I'd say it's fine for now.

**Comment by stefanv** [comment]
Merged #6855.

**Pull request closed without merging**